libc/cmake/modules/LLVMLibCHeaderRules.cmake

```diff
@@ -21,7 +21,11 @@ function(add_header target_name)
   set(absolute_path ${CMAKE_CURRENT_SOURCE_DIR}/${ADD_HEADER_HDR})
   file(RELATIVE_PATH relative_path ${LIBC_INCLUDE_SOURCE_DIR} ${absolute_path})
   set(dest_file ${LIBC_INCLUDE_DIR}/${relative_path})
-  set(src_file ${CMAKE_CURRENT_SOURCE_DIR}/${ADD_HEADER_HDR})
+  if(ADD_HEADER_SRCHDR)
```

Comment:
I think the whole project is kind of used to `HDR/HDRS/SRCS` are the source files.  Can you change the argument to something like `DEST_HDR`, and update the `dest_file`/`absolute_path` instead?